src/nwbinspector/checks/_nwbfile_metadata.py

```diff
@@ -352,3 +355,44 @@ def check_subject_id_no_slashes(subject: Subject) -> Optional[InspectorMessage]:
         )
 
     return None
+
+
+@register_check(importance=Importance.BEST_PRACTICE_SUGGESTION, neurodata_type=NWBFile)
+def check_file_extension(nwbfile: NWBFile) -> Optional[InspectorMessage]:
+    """
+    Check if the file extension contains ".nwb".
+    If a backend storage type is specified, check that it matches the backend storage type.
+
+    NWB files should use appropriate extensions based on their backend:
+    - .nwb (minimum recommendation), .nwb.h5 (HDF5), or .nwb.zarr (Zarr)
+
+    Best Practice: :ref:`best_practice_file_extension`
+    """
+    file_path = get_nwbfile_path_from_internal_object(nwbfile)
+
+    # Only perform the check if we can determine the file path
+    if file_path is not None:
+        file_extension = "".join(Path(file_path).suffixes)  # Get all suffixes for multi-part extensions
```

Comment:
```suggestion
        file_extension = "".join(Path(file_path).suffixes)  # Concatenate all suffixes for multi-part extensions
```

Adding this because I was confused as to what this does.